src/Type/Php/HashHmacFunctionsReturnTypeExtension.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+
+final class HashHmacFunctionsReturnTypeExtension implements DynamicFunctionReturnTypeExtension
+{
+
+	/** @var string[] */
+	private array $functions = [
+		'hash_hmac',
+		'hash_hmac_file',
+	];
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return in_array($functionReflection->getName(), $this->functions, true);
```

Comment:
This array can be here statically, it's not referenced anywhere else in the extension.

Reply:
Have inlined the value, hope this is correct?

I had orientated on other instances like `ArrayPointerFunctionsDynamicReturnTypeExtension` which also uses the `private array $functions = []` notation.